File: drills/elixir/solutions/nc46_merge_triplets__track_positions.exs
defmodule Solution do
  # Ask a different question: is each of the three positions hit exactly by some
  # usable triplet? The answer is yes exactly when all three are covered -- the
  # same condition, arrived at without taking maxima.
  def merge_triplets(triplets, {ta, tb, tc}) do
    usable = Enum.filter(triplets, fn {a, b, c} -> a <= ta and b <= tb and c <= tc end)

    Enum.any?(usable, fn {a, _, _} -> a == ta end) and
      Enum.any?(usable, fn {_, b, _} -> b == tb end) and
      Enum.any?(usable, fn {_, _, c} -> c == tc end)
  end
end

File: drills/python/solutions/nc46_merge_triplets__track_positions.py
def mergeTriplets(triplets, target):
    # Ask a different question: is each of the three positions hit exactly by
    # some usable triplet? The answer is yes exactly when all three are covered
    # -- the same condition, arrived at without taking maxima.
    usable = [t for t in triplets if all(t[i] <= target[i] for i in range(3))]
    return all(any(t[i] == target[i] for t in usable) for i in range(3))
